File: Examples/VisualBasic/WorkingWithCalendars/CreatingUpdatingAndRemoving/CreatingCalendar.vb
Imports System
Imports Aspose.Tasks
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports Aspose.Tasks.Examples.VisualBasic
Imports Aspose.Tasks.Saving

'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Tasks for .NET API reference 
'when the project is build. Please check https://docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Tasks for .NET API from http://www.aspose.com/downloads, 
'install it and then add its reference to this project. For any issues, questions or suggestions 
'please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'

Namespace WorkingWithCalendars.CreatingUpdatingAndRemoving
    Class CreatingCalendar
        Public Shared Sub Run()
            ' ExStart: CreatingCalendar
            ' Create a project instance
            Dim project As New Project()

            ' New calendar can be added to a project's calendar collection using the collection's Add method.
            Dim cal1 As Calendar = project.Calendars.Add("New Info")
            Dim cal2 As Calendar = project.Calendars.Add("no name")
            Dim cal3 As Calendar = project.Calendars.Add("cal3")

            ' Save the Project
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType.FullName)
            project.Save(dataDir & Convert.ToString("CreatingCalendar_out.Xml"), SaveFileFormat.XML)
            ' ExEnd: CreatingCalendar
        End Sub
    End Class
End Namespace

File: Examples/CSharp/WorkingWithCalendars/CreatingUpdatingAndRemoving/CreatingCalendar.cs
/*
This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Tasks for .NET API reference 
when the project is build. Please check https:// Docs.nuget.org/consume/nuget-faq for more information. 
If you do not wish to use NuGet, you can manually download Aspose.Tasks for .NET API from https://www.nuget.org/packages/Aspose.Tasks/, 
install it and then add its reference to this project. For any issues, questions or suggestions 
please feel free to contact us using https://forum.aspose.com/c/tasks
*/

namespace Aspose.Tasks.Examples.CSharp.WorkingWithCalendars.CreatingUpdatingAndRemoving
{
    internal class CreatingCalendar
    {
        public static void Run()
        {
            var dataDir = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod());

            //ExStart:CreatingCalendar
            //ExFor: CalendarCollection.Add(String)
            //ExFor: CalendarCollection.Add(String,Calendar)
            //ExSummary: Shows how to add new calendars.
            var project = new Project();

            // new calendars can be added to a project's calendar collection by using the collection's Add overloads.
            project.Calendars.Add("Calendar");
            var calendar = project.Calendars.Add("Parent");
            project.Calendars.Add("Child", calendar);

            project.Save(dataDir + "CreatingCalendar_out.Xml", Aspose.Tasks.Saving.SaveFileFormat.XML);
            //ExEnd:CreatingCalendar
        }
    }
}
